uninstall

```diff
@@ -0,0 +1,214 @@
+    self.symlink? ? dirname+readlink : self
+  end
+
+  def /(other)
+    self + other.to_s
+  end
+
+  def pretty_print
+    if self.symlink?
+      puts to_s + " -> " + resolved_path.to_s
+    elsif self.directory?
+      puts to_s + "/"
+    else
+      puts to_s
+    end
+  end
+end
+
+def ohai *args
+  puts "#{Tty.blue}==>#{Tty.white} #{args.shell_s}#{Tty.reset}"
+end
+
+def warn warning
+  puts "#{Tty.red}Warning#{Tty.reset}: #{warning.chomp}"
+end
+
+def system *args
+  warn "Failed during: #{args.shell_s}" unless Kernel.system(*args)
+end
+
+####################################################################### script
+
+$HOMEBREW_PREFIX_CANDIDATES=[]
+
+OptionParser.new do |opts|
+  opts.banner = "Homebrew Uninstaller\nUsage: ./uninstall [options]"
+  opts.summary_width = 16
+  opts.on("-pPATH", "--path=PATH", "Manually set path to Homebrew.") { |p| $HOMEBREW_PREFIX_CANDIDATES << Pathname.new(p) }
+  opts.on("--skip-cache-log", "Skipes removal of Caches and Logs files.") { |p| options[:skip_cache_log] = true }
+  opts.on("-f", "--force", "Forces removal of files, bypassing prompt. USE WITH CAUTION!") { options[:force] = true }
+  opts.on("-q", "--quiet", "Suppress all output.") { options[:quiet] = true }
+  opts.on("-d", "--dry-run", "Dry run, print results, but perform no actual operations.") { options[:dry_run] = true }
+  opts.on_tail("-h", "--help", "Display this message.") { puts opts; exit }
+end.parse!
+
+if $HOMEBREW_PREFIX_CANDIDATES.empty? # Attempt to locate Homebrew unless `--path` is passed
+  p = `brew --prefix` rescue ""
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new(p.strip) unless p.empty?
+  p = `command -v brew`
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new(p.strip).parent.parent unless p.empty?
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new("/usr/local") # Homebrew default path
+  $HOMEBREW_PREFIX_CANDIDATES << Pathname.new("#{ENV["HOME"]}/.linuxbrew") # Linuxbrew default path
+end
+
+$HOMEBREW_PREFIX = $HOMEBREW_PREFIX_CANDIDATES.detect do |p|
+  p.directory? && (p/"Library/Homebrew").directory?
+end
+
+abort "Failed to locate Homebrew" if $HOMEBREW_PREFIX.nil?
+
+$HOMEBREW_FILES = %W[
+  .git .gitignore .yardopts CODEOFCONDUCT.md CONTRIBUTING.md
+  LICENSE.txt README.md SUPPORTERS.md Cellar Library/.rubocop.yml
+  Library/Aliases Library/Contributions Library/ENV Library/Formula
+  Library/Homebrew Library/LinkedKegs Library/Taps Library/brew.rb
+  bin/brew share/doc/homebrew share/man/man1/brew.1
+].map { |p| $HOMEBREW_PREFIX/p }
+
+unless options[:skip_cache_log]
+  $HOMEBREW_FILES += %W[
+    #{ENV["HOME"]}/Library/Caches/Homebrew
+    #{ENV["HOME"]}/Library/Logs/Homebrew
+    /Library/Caches/Homebrew
+    #{ENV["HOME"]}/.cache/Homebrew
+  ].map { |p| Pathname.new(p) }
+end
+
+if /darwin/i === RUBY_PLATFORM
+  $HOMEBREW_FILES += %W[
+    /Applications
+    #{ENV["HOME"]}/Applications
+  ].map do |p|
+    Pathname.new(p).subdirs.select do |app|
+      app.resolved_path.to_s.start_with? $HOMEBREW_PREFIX.to_s
```

Comment:
It's possible for HOMEBREW_CELLAR to live outside HOMEBREW_PREFIX; I think this should be HOMEBREW_CELLAR